clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3210,6 +3210,11 @@ static void handleWorkGroupSizeHint(Sema &S, Decl *D, const ParsedAttr &AL) {
 
 void Sema::AddIntelReqdSubGroupSize(Decl *D, const AttributeCommonInfo &CI,
                                     Expr *E) {
+  if (checkAttrMutualExclusion<IntelNamedSubGroupSizeAttr>(*this, D, CI))
+    return;
+  if (checkAttrMutualExclusion<SYCLSimdAttr>(*this, D, CI))
```

Comment:
These will be needed in the merge function as well, otherwise we won't capture the issue when merging redeclarations.